components/esp_modem/include/generate/esp_modem_command_declare.inc

```diff
@@ -36,6 +37,7 @@
 #define BOOL_IN(param, name) const bool _ARG(param, name)
 #define BOOL_OUT(param, name) bool* _ARG(param, name)
 #define INT_OUT(param, name) int* _ARG(param, name)
+#define LIST_IN(param, name) const int* _ARG(param, name)
```

Comment:
Could we make the `LIST_IN` type array instead of ptr? (we should also call in *integer-list* or similar...)

Reply:
We probably could, I just wasn't sure exactly how to navigate the _ARG definitions. Although I think it might be better to use pointers and rather have _ARG take in pointer + size?

Reply:
Would this work?
```suggestion
#define LIST_IN(param, name) const int _ARG(param[], name[])
```
it still uses pointer and sizes, but is syntactically closer to an array.

but very minor point, feel free to resolve.

Reply:
If we want to make it more general, perhaps we could make it a `const void*` ?

Reply:
> If we want to make it more general, perhaps we could make it a const void* ? 

I personally prefer concrete types, so the list of integers looks good to me. Resolving, as I'm finding this okay.

Just to explain why I commented:
1) Types `int *` and `int []` are *almost* the same, but the latter suggest that it's an array (while the former is just a pointer)
2) Naming it `LIST_IN()` may suggest any kind of list, but it is a list of **integers** per your definition.

(Optionally you can just rename it to `INT_LIST_IN()` if that sounds good to you)